src/types/pages.ts

```diff
@@ -0,0 +1,118 @@
+import { Result } from "neverthrow"
+
+import { Brand, ToBrand } from "./util"
+
+type NameInfo = {
+  name: string
+  kind: string
+}
+
+type PageBrand<T extends NameInfo> = ToBrand<T, "name">
+
+export type ResourceRoomName = {
+  name: string & { __kind: "ResourceRoomName" }
+  kind: "ResourceRoomName"
+}
+
+export type SubcollectionPageName = {
+  name: string & { __kind: "SubcollectionPage" }
+  collection: string
+  subCollection: string
+  kind: "SubcollectionPage"
+}
+
+export type CollectionPageName = {
+  name: string & { __kind: "CollectionPage" }
+  collection: string
+  kind: "CollectionPage"
+}
+
+export type ContactUsPageName = {
+  name: string & { __kind: "ContactUsPage" }
+  kind: "ContactUsPage"
+}
+
+export type UnlinkedPageName = {
+  name: string & { __kind: "UnlinkedPage" }
+  kind: "UnlinkedPage"
+}
+
+export type HomepageName = {
+  name: string & { __kind: "Homepage" }
+  kind: "Homepage"
+}
+
+export type ResourceCategoryPageName = {
+  name: string & { __kind: "ResourceCategoryPage" }
+  resourceRoom: ToBrand<ResourceRoomName, "name">
+  resourceCategory: string
+  kind: "ResourceCategoryPage"
+}
+
+export type PageName =
+  | SubcollectionPageName
+  | CollectionPageName
+  | ContactUsPageName
+  | UnlinkedPageName
+  | HomepageName
+  | ResourceCategoryPageName
```

Comment:
Just for clarification, the usefulness of this banding of types will be amplified only when we add some level of validation checks when we actually do the casting right (though I understand this would be out of scope of this pr)?
Out of curiosity, is there a reason why you didnt chose to just add enums to differentiate the various types?

Reply:
i'm not entirely sure i understand the first question here @__@ the usefulness here is that it allows us to enforce a stronger API contract on subsequent methods going forward (eg: a method that operates on `Homepage` wouldn't accept an `UnlinkedPage`. Do take note that for the casting, it's only required for base types (string/number etc) as there's **no way to construct** a `string & {__kind: "blah"}`.

wrt why not use enums - because ts is structurally typed, we do need something to distinguish the types, esp if they're structurally identical. i just used a string (`kind`) cos i've ran into some issues w/ the implementation of enums in ts before wrt typing

Reply:
arh, I am curious, feel free to ignore this if you dont have concrete examples at hand, do you have examples of issues that you faced w/ implementation of enums in ts?